lib/metasploit/framework/compiler/golang.rb

```diff
@@ -0,0 +1,106 @@
+require 'open3'
+require 'fileutils'
+
+module Metasploit
+  module Framework
+    module Compiler
+      # Class for handling the go compiler
+      class Golang
+
+        UncompilablePayloadError = Class.new(StandardError)
+        CompiledPayloadNotFoundError = Class.new(StandardError)
+        GoModInitError = Class.new(StandardError)
+
+        DefaultCompiler = 'go'.freeze
+
+        attr_reader :outfile, :env, :ldflags, :compiler, :keep_src, :compiler_flags, :build_flags
+
+        def initialize(**opts)
+          @env = opts[:env]
+          @compiler = opts[:compiler] || DefaultCompiler
+          @compiler_flags = split(opts[:compiler_flags])
+          @outfile = opts[:outfile]
+          @build_flags = split(opts[:build_flags])
+          @ldflags = opts[:ldflags]
+          @keep_src = opts[:keep_src]
+        end
+
+        def self.available?
+          !!Msf::Util::Helper.which(DefaultCompiler)
+        end
+
+        def go_build_src(src, *files)
+          within_temp(files) do |dir, _fnames|
+            Tempfile.create(['main_', "__#{File.basename(outfile)}.go"], dir) do |file|
+              File.write(file, src)
+              if keep_src
+                src_copy = File.join(File.dirname(outfile), File.basename(file.path))
+                FileUtils.cp(file, src_copy)
+              end
+              return src_copy, compile
+            end
+          end
+        end
+
+        def go_build_files(*files)
+          within_temp(files) { compile }
+        end
+
+        def cmd_build
+          cmd = [compiler]
+          cmd << compiler_flags
+          cmd << 'build'
+          cmd << build_flags
```

Comment:
Since `compiler_flags` and `build_flags` are arrays, this should probably be:
```
          cmd += compiler_flags
          cmd << 'build'
          cmd += build_flags
```

That may fix the issues I was experiencing.